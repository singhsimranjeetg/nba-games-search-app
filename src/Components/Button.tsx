import styled, { css } from "styled-components";

const buttonStyles = css`
  display: flex;
  align-items: center;
  padding: 0.5rem 1rem;
  border-radius: 100px;
  border: none;
  transition: box-shadow 300ms ease;
  cursor: pointer;
  justify-content: center;
  color: var(--white);
  width: min-content;
  user-select: none;

  &:disabled {
    background-color: var(--surface);
  }

  &:focus {
    outline: none;
  }

  @media (min-width: 769px) {
    &:hover {
      box-shadow: var(--hover);
    }
  }
`;

const PrimaryButton = styled.button`
  ${buttonStyles}
  background-color: var(--green);
`;

const SecondaryButton = styled.button`
  ${buttonStyles}
  background-color: var(--black);
`;

export type ButtonProps = {
  secondary?: boolean;
  disabled?: boolean;
  children?: React.ReactFragment | HTMLCollection | string;
  onClick: (e?: React.MouseEvent) => void;
} & React.ButtonHTMLAttributes<HTMLButtonElement>;

const Button = ({
  secondary,
  disabled,
  onClick,
  children,
  style,
  id,
  ...other
}: ButtonProps) =>
  secondary ? (
    <SecondaryButton
      style={style}
      disabled={disabled}
      onClick={onClick}
      id={id}
      {...other}
    >
      <strong>{children}</strong>
    </SecondaryButton>
  ) : (
    <PrimaryButton
      style={style}
      disabled={disabled}
      onClick={onClick}
      id={id}
      {...other}
    >
      <strong>{children}</strong>
    </PrimaryButton>
  );

export default Button;
