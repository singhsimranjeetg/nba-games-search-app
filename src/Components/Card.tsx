import styled from "styled-components";

const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 1rem;
  margin: 1rem 0;
  border: solid 1px var(--surface);
  transition: box-shadow 300ms ease;
  border-radius: 0.5rem;
  height: min-content;
  background-color: var(--white);

  & label:not(.detail) {
    font-weight: 600;
  }

  @media (max-width: 768px) {
    width: 100%;
  }

  @media (min-width: 769px) {
    &:hover {
      box-shadow: var(--hover);
    }
  }
`;

interface CardProps {
  children?: React.ReactFragment | HTMLCollection | string;
  style?: React.CSSProperties | undefined;
}

const Card = ({ children, style }: CardProps) => (
  <CardContainer style={style}>{children}</CardContainer>
);

export default Card;
