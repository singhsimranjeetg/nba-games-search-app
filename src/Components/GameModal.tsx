import styled from 'styled-components';
import { Modal } from '@shopify/polaris';
import GameDetails from '../Components/GameDetails.component';

const GameCardContainer = styled.div`
  display: flex;
  flex-direction: row;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;

interface GameModalProps {
  game: any;
  open: boolean;
  handleModalChange: (e?: React.MouseEvent) => void;
}

const GameModal = ({ game, open, handleModalChange }: GameModalProps) => {
  return (
    <Modal title='Game Details' onClose={handleModalChange} open={open}>
      <Modal.Section>
        <GameCardContainer>
          <GameDetails game={game} key={game} />
        </GameCardContainer>
      </Modal.Section>
      <Modal.Section></Modal.Section>
    </Modal>
  );
};

export default GameModal;
