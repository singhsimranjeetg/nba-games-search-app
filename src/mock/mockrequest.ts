import { GamesData } from "./gamesByDate";

export const searchMock = (
  input: RequestInfo,
  init?: RequestInit | undefined
): Promise<Response> =>
  new Promise((resolve) =>
    resolve({
      json: () => Promise.resolve(GamesData),
    } as Response)
  );