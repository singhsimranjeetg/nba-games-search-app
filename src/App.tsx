import { useState } from 'react';
import styled from 'styled-components';
import Card from './Components/Card';
import { GlobalStyles } from './global.styles';
import { AppProvider, Autocomplete } from '@shopify/polaris';
import enTranslations from '@shopify/polaris/locales/en.json';
import SearchMajor from './assets/SearchMajor.svg';
import { ListAction } from './util/interfaces';
import GameComponent from './Components/Game';

// Mock Data
import { GamesData } from './mock/gamesByDate';

//real
// const API_KEY = process.env.REACT_APP_API_KEY;
// const HOST_URL = `api-nba-v1.p.rapidapi.com`;

const AppContainer = styled.div`
  display: flex;
  height: 100%;
`;

const ContentContainer = styled.div`
  position: relative;
  display: flex;
  padding: 2rem 1rem;
  flex-direction: column;
  max-width: 1080px;
  align-items: center;

  @media (min-width: 769px) {
    width: 70%;
    margin: auto;
  }

  @media (max-width: 768px) {
    width: 100%;
  }
`;

const HeaderContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 1rem 0;
`;

const SelectionContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 1rem 0;
  justify-content: flex-start;
`;

const CardsContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

const SearchContainer = styled.div`
  display: flex;
  flex-direction: column;

  & button {
    margin-left: auto;
    margin-right: 0;
  }

  @media (max-width: 768px) {
    width: 100%;
  }

  @media (min-width: 769px) {
    min-width: 480px;
    margin-right: 2rem;
    width: calc(60% - 2rem);
  }
`;

const ListContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

const SearchIcon = styled.img`
  height: 24px;
  width: 24px;
  vertical-align: middle;
`;

const ActionsContainer = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  padding: 1rem 0;
  justify-content: flex-end;
  align-items: center;

  & > button:not(:first-child) {
    margin-left: 1.5rem;
  }
`;

const Button = styled.button`
  font-size: 16px;
  font-weight: 600;
  color: #fff;
  padding: 10px 20px;
  border-radius: 20px;
  border: none;
  background: #1fae51;

  &:hover {
    cursor: pointer;
    background: #1fae30;
  }
`;

const GameInstructions = () => (
  <label className='detail'>
    click on a game to get more details or scorecard
  </label>
);

const App = () => {
  const [searchResults, setSearchResults] = useState<any>('');
  const [searchQuery, setSearchQuery] = useState('');
  const [activeSearchQuery, setActiveSearchQuery] = useState('');

  const getGamesByDate = async (date: string) => {
    //real
    // fetch(`https://${HOST_URL}/games/date/${date}`, {
    //   method: 'GET',
    //   headers: {
    //     'x-rapidapi-host': `${HOST_URL}`,
    //     'x-rapidapi-key': `${API_KEY}`,
    //   },
    // })
    //   .then(async (response) => {
    //     const results = await response.json();
    //     if (results) {
    //       // const formattedList = formatMovies(results.api.games);
    //       setSearchResults(results.api.games);
    //       setActiveSearchQuery(searchQuery);
    //     } else {
    //       setBannerText('No games related to that date were found!');
    //       clearSearch();
    //     }

    //     // setGameDetails(results.api.game);
    //   })
    //   .catch((err) => {
    //     console.error(err);
    //   });
    setSearchResults(GamesData);
    setActiveSearchQuery(searchQuery);
    console.log('searchresults', searchResults);
  };

  interface GameListProps {
    games: any;
    activeDisabled?: boolean;
    action: ListAction;
  }
  const GameList = ({
    games,
    action,
    activeDisabled = false,
  }: GameListProps) => (
    <div>
      {games.map((game: any) => (
        <GameComponent
          key={game.gameId}
          action={action}
          game={game}
          disabled={activeDisabled}
        ></GameComponent>
      ))}
    </div>
  );

  return (
    <AppProvider i18n={enTranslations}>
      <AppContainer>
        <GlobalStyles />
        <ContentContainer>
          <SelectionContainer>
            <HeaderContainer>
              <h1>NBA Games Search App</h1>
            </HeaderContainer>
            <label>Search games by date using the date picker below.</label>

            <CardsContainer>
              <SearchContainer>
                <Card>
                  <Autocomplete.TextField
                    onChange={(e) => setSearchQuery(e)}
                    label='Games Date'
                    value={searchQuery}
                    prefix={<SearchIcon alt='search icon' src={SearchMajor} />}
                    type='date'
                  />
                  <ActionsContainer>
                    <Button
                      onClick={() => getGamesByDate(searchQuery)}
                      id='search-button'
                    >
                      Search
                    </Button>
                  </ActionsContainer>
                </Card>
                <Card>
                  <label>
                    <b>
                      Games results{' '}
                      {activeSearchQuery
                        ? `for ${' '} ${activeSearchQuery}`
                        : 'will appear here'}
                    </b>
                  </label>
                  {activeSearchQuery && (
                    <ListContainer id='search-results'>
                      <GameInstructions />
                      <GameList
                        games={searchResults}
                        action='ADD'
                        activeDisabled
                      />
                    </ListContainer>
                  )}
                </Card>
              </SearchContainer>
            </CardsContainer>
          </SelectionContainer>
        </ContentContainer>
      </AppContainer>
    </AppProvider>
  );
};

export default App;
