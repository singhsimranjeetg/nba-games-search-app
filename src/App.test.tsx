import { shallow, render } from 'enzyme';
import React from 'react';

import App from './App';
import GameDetails from './Components/GameDetails.component';
import GameComponent from './Components/Game';
import GameModal from './Components/GameModal';
import Button from './Components/Button';
import Card from './Components/Card';

//mock
import { GameData } from './mock/gamedata.mock';

describe('rendering components', () => {
  it('expect to render App Component', () => {
    shallow(<App />);
  });
  it('expect to render Game Component', () => {
    shallow(<GameComponent game={GameData.api.game[0]} />);
  });
  it('expect to render GameDetails Component', () => {
    shallow(<GameDetails game={GameData.api.game[0]} />);
  });
  it('expect to render GameModal Component', () => {
    shallow(
      <GameModal
        game={GameData.api.game[0]}
        open
        handleModalChange={() => {}}
      />
    );
  });
  it('expect to render Button Component', () => {
    shallow(<Button onClick={() => {}} />);
  });
  it('expect to render Card Component', () => {
    shallow(<Card />);
  });
});

describe('Test <App />', () => {
  it('renders a search button', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('#search-button').exists()).toBeTruthy();
  });
});

//snapshot tests
it('App', () => {
  expect(shallow(<App />)).toMatchSnapshot();
});
