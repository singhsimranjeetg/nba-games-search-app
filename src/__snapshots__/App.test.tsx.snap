// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<AppProvider$1
  i18n={
    Object {
      "Polaris": Object {
        "ActionMenu": Object {
          "RollupActions": Object {
            "rollupButton": "Actions",
          },
        },
        "Actions": Object {
          "moreActions": "More actions",
        },
        "Autocomplete": Object {
          "spinnerAccessibilityLabel": "Loading",
        },
        "Avatar": Object {
          "label": "Avatar",
          "labelWithInitials": "Avatar with initials {initials}",
        },
        "Badge": Object {
          "PROGRESS_LABELS": Object {
            "complete": "Complete",
            "incomplete": "Incomplete",
            "partiallyComplete": "Partially complete",
          },
          "STATUS_LABELS": Object {
            "attention": "Attention",
            "critical": "Critical",
            "info": "Info",
            "new": "New",
            "success": "Success",
            "warning": "Warning",
          },
          "progressAndStatus": "{statusLabel} {progressLabel}",
        },
        "Button": Object {
          "connectedDisclosureAccessibilityLabel": "Related actions",
          "spinnerAccessibilityLabel": "Loading",
        },
        "Common": Object {
          "cancel": "Cancel",
          "checkbox": "checkbox",
          "clear": "Clear",
          "close": "Close",
          "more": "More",
          "newWindowAccessibilityHint": "(opens a new window)",
          "submit": "Submit",
          "undo": "Undo",
        },
        "ContextualSaveBar": Object {
          "discard": "Discard",
          "save": "Save",
        },
        "DataTable": Object {
          "navAccessibilityLabel": "Scroll table {direction} one column",
          "sortAccessibilityLabel": "sort {direction} by",
          "totalRowHeading": "Total",
          "totalsRowHeading": "Totals",
        },
        "DatePicker": Object {
          "days": Object {
            "friday": "Friday",
            "monday": "Monday",
            "saturday": "Saturday",
            "sunday": "Sunday",
            "thursday": "Thursday",
            "tuesday": "Tuesday",
            "wednesday": "Wednesday",
          },
          "daysAbbreviated": Object {
            "friday": "Fr",
            "monday": "Mo",
            "saturday": "Sa",
            "sunday": "Su",
            "thursday": "Th",
            "tuesday": "Tu",
            "wednesday": "We",
          },
          "end": "End of range",
          "months": Object {
            "april": "April",
            "august": "August",
            "december": "December",
            "february": "February",
            "january": "January",
            "july": "July",
            "june": "June",
            "march": "March",
            "may": "May",
            "november": "November",
            "october": "October",
            "september": "September",
          },
          "nextMonth": "Show next month, {nextMonth} {nextYear}",
          "previousMonth": "Show previous month, {previousMonthName} {showPreviousYear}",
          "start": "Start of range",
          "today": "Today ",
        },
        "DiscardConfirmationModal": Object {
          "message": "If you discard changes, you’ll delete any edits you made since you last saved.",
          "primaryAction": "Discard changes",
          "secondaryAction": "Continue editing",
          "title": "Discard all unsaved changes",
        },
        "DropZone": Object {
          "FileUpload": Object {
            "actionHintFile": "or drop files to upload",
            "actionHintImage": "or drop images to upload",
            "actionTitleFile": "Add file",
            "actionTitleImage": "Add image",
            "label": "Upload file",
          },
          "errorOverlayTextFile": "File type is not valid",
          "errorOverlayTextImage": "Image type is not valid",
          "overlayTextFile": "Drop file to upload",
          "overlayTextImage": "Drop image to upload",
        },
        "EmptySearchResult": Object {
          "altText": "Empty search results",
        },
        "Filters": Object {
          "cancel": "Cancel",
          "clear": "Clear",
          "clearAllFilters": "Clear all filters",
          "clearLabel": "Clear {filterName}",
          "done": "Done",
          "filter": "Filter {resourceName}",
          "moreFilters": "More filters",
          "moreFiltersWithCount": "More filters ({count})",
          "noFiltersApplied": "No filters applied",
        },
        "Frame": Object {
          "Navigation": Object {
            "closeMobileNavigationLabel": "Close navigation",
          },
          "skipToContent": "Skip to content",
        },
        "Icon": Object {
          "backdropWarning": "The {color} icon doesn’t accept backdrops. The icon colors that have backdrops are: {colorsWithBackDrops}",
        },
        "Loading": Object {
          "label": "Page loading bar",
        },
        "MediaCard": Object {
          "popoverButton": "Actions",
        },
        "Modal": Object {
          "iFrameTitle": "body markup",
          "modalWarning": "These required properties are missing from Modal: {missingProps}",
        },
        "Pagination": Object {
          "next": "Next",
          "pagination": "Pagination",
          "previous": "Previous",
        },
        "ProgressBar": Object {
          "exceedWarningMessage": "Values passed to the progress prop shouldn’t exceed 100. Setting {progress} to 100.",
          "negativeWarningMessage": "Values passed to the progress prop shouldn’t be negative. Resetting {progress} to 0.",
        },
        "ResourceList": Object {
          "BulkActions": Object {
            "actionsActivatorLabel": "Actions",
            "moreActionsActivatorLabel": "More actions",
            "warningMessage": "To provide a better user experience. There should only be a maximum of {maxPromotedActions} promoted actions.",
          },
          "DateSelector": Object {
            "FilterLabelForValue": Object {
              "coming_month": "next month",
              "coming_quarter": "in the next 3 months",
              "coming_week": "next week",
              "coming_year": "in the next year",
              "on_or_after": "after {date}",
              "on_or_before": "before {date}",
              "past_month": "in the last month",
              "past_quarter": "in the last 3 months",
              "past_week": "in the last week",
              "past_year": "in the last year",
            },
            "SelectOptions": Object {
              "ComingMonth": "next month",
              "ComingQuarter": "in the next 3 months",
              "ComingWeek": "next week",
              "ComingYear": "in the next year",
              "OnOrAfter": "on or after",
              "OnOrBefore": "on or before",
              "PastMonth": "in the last month",
              "PastQuarter": "in the last 3 months",
              "PastWeek": "in the last week",
              "PastYear": "in the last year",
            },
            "dateFilterLabel": "Select a value",
            "dateValueError": "Match YYYY-MM-DD format",
            "dateValueLabel": "Date",
            "dateValuePlaceholder": "YYYY-MM-DD",
          },
          "FilterControl": Object {
            "textFieldLabel": "Search {resourceNamePlural}",
          },
          "FilterCreator": Object {
            "addFilterButtonLabel": "Add filter",
            "filterButtonLabel": "Filter",
            "selectFilterKeyPlaceholder": "Select a filter…",
            "showAllWhere": "Show all {resourceNamePlural} where:",
          },
          "FilterValueSelector": Object {
            "selectFilterValuePlaceholder": "Select a filter…",
          },
          "Item": Object {
            "actionsDropdown": "Actions dropdown",
            "actionsDropdownLabel": "Actions for {accessibilityLabel}",
            "viewItem": "View details for {itemName}",
          },
          "a11yCheckboxDeselectAllMultiple": "Deselect all {itemsLength} {resourceNamePlural}",
          "a11yCheckboxDeselectAllSingle": "Deselect {resourceNameSingular}",
          "a11yCheckboxSelectAllMultiple": "Select all {itemsLength} {resourceNamePlural}",
          "a11yCheckboxSelectAllSingle": "Select {resourceNameSingular}",
          "allFilteredItemsSelected": "All {itemsLength}+ {resourceNamePlural} in this filter are selected.",
          "allItemsSelected": "All {itemsLength}+ {resourceNamePlural} in your store are selected.",
          "ariaLivePlural": "{itemsLength} items",
          "ariaLiveSingular": "{itemsLength} item",
          "defaultItemPlural": "items",
          "defaultItemSingular": "item",
          "emptySearchResultDescription": "Try changing the filters or search term",
          "emptySearchResultTitle": "No {resourceNamePlural} found",
          "loading": "Loading {resource}",
          "selectAllFilteredItems": "Select all {itemsLength}+ {resourceNamePlural} in this filter",
          "selectAllItems": "Select all {itemsLength}+ {resourceNamePlural} in your store",
          "selectButtonText": "Select",
          "selected": "{selectedItemsCount} selected",
          "showing": "Showing {itemsCount} {resource}",
          "showingTotalCount": "Showing {itemsCount} of {totalItemsCount} {resource}",
          "sortingLabel": "Sort by",
        },
        "SkeletonPage": Object {
          "loadingLabel": "Page loading",
        },
        "Spinner": Object {
          "warningMessage": "The color {color} is not meant to be used on {size} spinners. The colors available on large spinners are: {colors}",
        },
        "Tabs": Object {
          "toggleTabsLabel": "More tabs",
        },
        "Tag": Object {
          "ariaLabel": "Remove {children}",
        },
        "TextField": Object {
          "characterCount": "{count} characters",
          "characterCountWithMaxLength": "{count} of {limit} characters used",
        },
        "TooltipOverlay": Object {
          "accessibilityLabel": "Tooltip: {label}",
        },
        "TopBar": Object {
          "SearchField": Object {
            "clearButtonLabel": "Clear",
            "search": "Search",
          },
          "toggleMenuLabel": "Toggle menu",
        },
        "VideoThumbnail": Object {
          "playButtonA11yLabel": Object {
            "default": "Play video",
            "defaultWithDuration": "Play video of length {duration}",
            "duration": Object {
              "hours": Object {
                "one": Object {
                  "andMinute": "{hourCount} hour and {minuteCount} minute",
                  "andMinutes": "{hourCount} hour and {minuteCount} minutes",
                  "andSecond": "{hourCount} hour and {secondCount} second",
                  "andSeconds": "{hourCount} hour and {secondCount} seconds",
                  "minuteAndSecond": "{hourCount} hour, {minuteCount} minute, and {secondCount} second",
                  "minuteAndSeconds": "{hourCount} hour, {minuteCount} minute, and {secondCount} seconds",
                  "minutesAndSecond": "{hourCount} hour, {minuteCount} minutes, and {secondCount} second",
                  "minutesAndSeconds": "{hourCount} hour, {minuteCount} minutes, and {secondCount} seconds",
                  "only": "{hourCount} hour",
                },
                "other": Object {
                  "andMinute": "{hourCount} hours and {minuteCount} minute",
                  "andMinutes": "{hourCount} hours and {minuteCount} minutes",
                  "andSecond": "{hourCount} hours and {secondCount} second",
                  "andSeconds": "{hourCount} hours and {secondCount} seconds",
                  "minuteAndSecond": "{hourCount} hours, {minuteCount} minute, and {secondCount} second",
                  "minuteAndSeconds": "{hourCount} hours, {minuteCount} minute, and {secondCount} seconds",
                  "minutesAndSecond": "{hourCount} hours, {minuteCount} minutes, and {secondCount} second",
                  "minutesAndSeconds": "{hourCount} hours, {minuteCount} minutes, and {secondCount} seconds",
                  "only": "{hourCount} hours",
                },
              },
              "minutes": Object {
                "one": Object {
                  "andSecond": "{minuteCount} minute and {secondCount} second",
                  "andSeconds": "{minuteCount} minute and {secondCount} seconds",
                  "only": "{minuteCount} minute",
                },
                "other": Object {
                  "andSecond": "{minuteCount} minutes and {secondCount} second",
                  "andSeconds": "{minuteCount} minutes and {secondCount} seconds",
                  "only": "{minuteCount} minutes",
                },
              },
              "seconds": Object {
                "one": "{secondCount} second",
                "other": "{secondCount} seconds",
              },
            },
          },
        },
      },
    }
  }
>
  <styled.div>
    <Memo(l) />
    <styled.div>
      <styled.div>
        <styled.div>
          <h1>
            NBA Games Search App
          </h1>
        </styled.div>
        <label>
          Search games by date using the date picker below.
        </label>
        <styled.div>
          <styled.div>
            <Card>
              <TextField$1
                label="Games Date"
                onChange={[Function]}
                prefix={
                  <styled.img
                    alt="search icon"
                    src="SearchMajor.svg"
                  />
                }
                type="date"
                value=""
              />
              <styled.div>
                <styled.button
                  id="search-button"
                  onClick={[Function]}
                >
                  Search
                </styled.button>
              </styled.div>
            </Card>
            <Card>
              <label>
                <b>
                  Games results
                   
                  will appear here
                </b>
              </label>
            </Card>
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</AppProvider$1>
`;
